### ISSUE TITLE
Linux: Pre-built SDK engine cannot be built on certain machines

### ISSUE THREAD

**Issue by LB-ArturZieba** [o3de/o3de#15140]
**Describe the bug**
Pre-built SDK engine fails building on Linux on certain machines (issue was reproduced on two different machines with AMD/NVIDIA CPU/GPU, while it was not reproduced on Intel/NVIDIA machine), for details please refer to the **Hardinfo files** section.

Error printed to the Terminal:
```
[2071/2354] Building CXX object External/RemoteTools-b18f3590/Code/CMakeFiles/RemoteTools.Private.Static.dir/profile/Unity/unity_0_cxx.cxx.o
FAILED: External/RemoteTools-b18f3590/Code/CMakeFiles/RemoteTools.Private.Static.dir/profile/Unity/unity_0_cxx.cxx.o 
/usr/bin/clang++-14 -DAZ_BUILD_CONFIGURATION_TYPE=\"profile\" -DAZ_ENABLE_DEBUG_TOOLS -DAZ_ENABLE_TRACING -DAZ_PROFILE_BUILD -DLINUX -DLINUX64 -DNDEBUG -DPAL_TRAIT_LINUX_WINDOW_MANAGER_XCB -D_FORTIFY_SOURCE=2 -D_HAS_EXCEPTIONS=0 -D_PROFILE -D__linux__ -DCMAKE_INTDIR=\"profile\" -I/home/gpc-610/builds/o3de/Gems/RemoteTools/Code/Include -I/home/gpc-610/builds/o3de/Gems/RemoteTools/Code/Source -I/home/gpc-610/builds/o3de/build/linux/External/RemoteTools-b18f3590/Code/Azcg/Generated/RemoteTools.Private.Static -I/home/gpc-610/builds/o3de/Code/Framework/AzCore/. -I/home/gpc-610/builds/o3de/Code/Framework/AzCore/Platform/Linux -I/home/gpc-610/builds/o3de/Code/Framework/AzCore/Platform/Common -I/home/gpc-610/builds/o3de/Code/Framework/AzFramework/. -I/home/gpc-610/builds/o3de/Code/Framework/AzFramework/Platform/Linux -I/home/gpc-610/builds/o3de/Code/Framework/AzFramework/Platform/Common/Xcb -I/home/gpc-610/builds/o3de/Code/Framework/AzNetworking/. -I/home/gpc-610/builds/o3de/Code/Framework/AzNetworking/Platform/Common -I/home/gpc-610/builds/o3de/Code/Framework/AzNetworking/Platform/Linux -I/home/gpc-610/builds/o3de/build/linux/Code/Framework/AzNetworking/Azcg/Generated/AzNetworking -isystem /home/gpc-610/o3de-packages/packages/Lua-5.4.4-rev1-linux/Lua/include -isystem /home/gpc-610/o3de-packages/packages/RapidJSON-1.1.0-rev1-multiplatform/RapidJSON/include -isystem /home/gpc-610/o3de-packages/packages/RapidXML-1.13-rev1-multiplatform/RapidXML/include -isystem /home/gpc-610/o3de-packages/packages/zlib-1.2.11-rev5-linux/zlib/include -isystem /home/gpc-610/o3de-packages/packages/zstd-1.35-multiplatform/zstd/lib -isystem /home/gpc-610/o3de-packages/packages/cityhash-1.1-multiplatform/cityhash/src -isystem /home/gpc-610/o3de-packages/packages/lz4-1.9.3-vcpkg-rev4-linux/lz4/lz4/include -fno-exceptions -fvisibility=hidden -fvisibility-inlines-hidden -Wall -Werror -Wno-inconsistent-missing-override -Wrange-loop-analysis -Wno-unknown-warning-option -Wno-parentheses -Wno-reorder -Wno-switch -Wno-undefined-var-template -msse4.1  -O2 -g -fstack-protector-all -fstack-check -fPIC -std=c++17 -MD -MT External/RemoteTools-b18f3590/Code/CMakeFiles/RemoteTools.Private.Static.dir/profile/Unity/unity_0_cxx.cxx.o -MF External/RemoteTools-b18f3590/Code/CMakeFiles/RemoteTools.Private.Static.dir/profile/Unity/unity_0_cxx.cxx.o.d -o External/RemoteTools-b18f3590/Code/CMakeFiles/RemoteTools.Private.Static.dir/profile/Unity/unity_0_cxx.cxx.o -c /home/gpc-610/builds/o3de/build/linux/External/RemoteTools-b18f3590/Code/CMakeFiles/RemoteTools.Private.Static.dir/Unity/unity_0_cxx.cxx
In file included from /home/gpc-610/builds/o3de/build/linux/External/RemoteTools-b18f3590/Code/CMakeFiles/RemoteTools.Private.Static.dir/Unity/unity_0_cxx.cxx:3:
/home/gpc-610/builds/o3de/Gems/RemoteTools/Code/Source/RemoteToolsSystemComponent.cpp:568:14: error: variable 'hasActiveConnection' set but not used [-Werror,-Wunused-but-set-variable]
        bool hasActiveConnection = false;
             ^
1 error generated.
[2078/2354] Building CXX object External/ScriptCanvas-3b33b874/Code/CMakeFiles/ScriptCanvas.Static.dir/profile/Unity/unity_11_cxx.cxx.o
ninja: build stopped: subcommand failed.
gpc-610@GPC-610:~/builds/o3de$
```

It should be noted that https://github.com/o3de/o3de/commit/7993300824056286755c83bd838b1afe7777d39f could be related to this issue.

It should also be noted that this issue does not occur on Source build.

**Steps to reproduce**
1. Configure and build a Pre-built SDK engine.

**Expected behavior**
Engine is built.

**Actual behavior**
Engine build fails due to an error.

**Hardinfo files**
Issue not reproduced on Stabilization/2305 (9e5047f) and Development (d6ef267):
[GPC-464_hardinfo_report.txt](https://github.com/o3de/o3de/files/10991605/GPC-464_hardinfo_report.txt) - Intel i5-9600K CPU + NVIDIA RTX 2060

Issue reproduced on Pre-built SDK Stabilization/2305 (9e5047f):
[GPC-606_hardinfo_report.txt](https://github.com/o3de/o3de/files/10991607/GPC-606_hardinfo_report.txt) - AMD Ryzen 5 3600 + NVIDIA RTX 2060 SUPER
[GPC-610_hardinfo_report.txt](https://github.com/o3de/o3de/files/10991610/GPC-610_hardinfo_report.txt) - AMD Ryzen 5 3600 + NVIDIA RTX 2060 SUPER

**Found in Branch**
Stabilization/2305 (9e5047f)

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**
9e5047fef0a61cfb3044342f0f1fd468366dcf7f

**Desktop**
 - Device: PC
 - OS: Linux
 - Version: Ubuntu 22.04.1 LTS
 - CPU AMD Ryzen 5 3600
 - GPU NVIDIA GeForce RTX 2060 SUPER
 - Memory 16GB

**Comment by LB-ArturZieba** [comment]
It should be noted that one of the machines (GPC-606 in the Hardinfo files section) was unsuccessful in building a new project on the Source engine due to this issue, while another was able to launch Editor (GPC-610 in the Hardinfo files section).

**Comment by amzn-changml** [comment]
@o3de/sig-testing to confirm that this remains an issue

**Comment by LB-ArturZieba** [comment]
Issue verified as fixed on the Stabilization/2305 (d28da8f) build. We have reverified this on the same machines mentioned before (GPC-606 and GPC-610) with Source AutomatedTesting project and Pre-built SDK (on which Default Template, Default Template + Remote Tools Connection and All Gems projects were built and successfully launched) respectively.
